askama_parser/src/node.rs

```diff
@@ -128,10 +128,23 @@ pub enum Target<'a> {
     CharLit(&'a str),
     BoolLit(&'a str),
     Path(Vec<&'a str>),
+    OrChain(Vec<Target<'a>>),
 }
 
 impl<'a> Target<'a> {
+    /// Parses multiple targets with `or` separating them
     pub(super) fn parse(i: &'a str) -> ParseResult<'a, Self> {
+        map(
+            separated_list1(ws(tag("or")), Self::parse_one),
+            |opts| match <[_; 1]>::try_from(opts) {
```

Comment:
I don't find this `<[_; 1]>::try_from()` idiom particularly clear. How about this:

```rust
match opts.len() {
    1 => opts.pop().unwrap(),
    _ => Self::OrChain(opts),
}
```

(Is this actually necessary? What happens if we just always yield a `Vec<Target<'a>>` from `parse()`?)

Reply:
I liked the `try_from` because it avoided `mut` and `.unwrap()`, but I can change this.
The `match` statement helps avoid one item `Target::OrChain` which will help debugging by cleaning up the output of `println!("{target:#?}")`.